EarthAsDMAnalyzer/plugins/EarthAsDMAnalyzer.cc

```diff
@@ -580,6 +690,93 @@ void EarthAsDMAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup&
     if (dtSeg_n > 0) {
       muonAvgEtaFromDTseg = muonSumEtaFromDTseg / dtSeg_n;
       muonAvgPhiFromDTseg = muonSumPhiFromDTseg / dtSeg_n;
+      
+      float rSquared = calculateRSquared( dtGlobalPointYValues, t0timingValues);
+      // Store the R-squared value in the rSquaredValues vector
+      rSquaredValues_.push_back( rSquared);
+
+      float PearsonCorrelation = pearsonCorrelation(dtGlobalPointYValues, t0timingValues);
+      PearsonValues_.push_back( PearsonCorrelation);
+      if (PearsonCorrelation > 0.9) {
+        // cout << "This track is bad 2" << endl;
+        // cout << "Station Vector: ";
+        // for (size_t i = 0; i < stationVector_.size(); ++i) {
+        //     cout << "Station " << stationVector_[i] << " in Sector " << sectorVector_[i] << " ";
+        for (size_t i = 0; i < stationVector_.size(); ++i) {
+          stationSectorVector_.push_back(std::make_pair(stationVector_[i], sectorVector_[i]));
+        }
+        // Print the combined vector
```

Comment:
these kinda cuts should happen later, not when you create the ntuple. For example what is you want to see this higher than 0.9 ? you'd need to rerun this